viewer/packages/api/src/api-helpers/Image360ApiHelper.ts

```diff
@@ -84,6 +95,23 @@ export class Image360ApiHelper {
     ) {
       this.exit360Image();
     }
+
+    const imageCollectionToRemove: DefaultImage360Collection[] = [];
+    this._imageCollection.forEach(imageCollection => {
+      pullAll(
+        imageCollection.image360Entities,
+        entities.map(entity => entity as Image360Entity)
+      );
+      if (imageCollection.image360Entities.length === 0) {
+        imageCollection.dispose();
+        imageCollectionToRemove.push(imageCollection);
+      }
+    });
+
+    if (imageCollectionToRemove.length > 0) {
+      pullAll(this._imageCollection, imageCollectionToRemove);
+    }
```

Comment:
```suggestion
    this._imageCollection.forEach(imageCollection => pullAll(
        imageCollection.image360Entities,
        entities.map(entity => entity as Image360Entity)
      );
    );
    
    const imageCollectionsToRemove = imageCollection.filter(collection => collection.image360Entities.length === 0);
    imageCollectionsToRemove.forEach(collection => collection.dispose());
    pullAll(this._imageCollection, imageCollectionsToRemove);
```

Reply:
note that I haven't tested this and it probably has linting errors, but I think this is cleaner, do you agree?